### PR TITLE
Notifications + General Styling

### PR DESCRIPTION
**Pull request by abhikaboy**
## Description

[Link to Ticket](insert the link to your ticket inside the parenthesis here)

## Checklist:
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] Some tests are included that test my code

### CODE REVIEW

File: backend/internal/models/user.go [hunk 1]
```diff
@@ -13,3 +13,14 @@ type User struct {
 	CreatedAt      time.Time `json:"created_at"`
 	UpdatedAt      time.Time `json:"updated_at"`
 }
+
+type Notification struct {
+	ID               int       `json:"id"`
+	Thumbnail        string    `json:"thumbnail"` // Url of the notification thumbnail
```
Comment (ddusichka) [inline backend/internal/models/user.go:19]: where do the thumbnails come from? like what are the possible values

File: backend/internal/models/user.go [hunk 2]
```diff
@@ -13,3 +13,14 @@ type User struct {
...
+	CreatedAt        time.Time `json:"created_at"`
+	Type             string    `json:"type"`        // Type of the notifcation
+	RecieverID       string    `json:"reciever_id"` // Reciever of the notification
```
Comment (ddusichka) [inline backend/internal/models/user.go:22]: apologies it's the spelling bee nerd in me

```suggestion
	ReceiverID       string    `json:"receiver_id"` // Receiver of the notification
```

File: backend/internal/storage/postgres/schema/user.go [hunk 3]
```diff
@@ -88,6 +88,30 @@ func (r *UserRepository) Follow(ctx context.Context, follower uuid.UUID, followi
 	if err != nil {
 		return false, err
 	}
+	println("inserted into follower table")
```
Comment (ddusichka) [inline backend/internal/storage/postgres/schema/user.go:91]:
```suggestion
```

File: backend/internal/storage/postgres/schema/user.go [hunk 4]
```diff
@@ -88,6 +88,30 @@ func (r *UserRepository) Follow(ctx context.Context, follower uuid.UUID, followi
...
+
+	// get the username and profile picture of the followee
+	var followerUsername string
+	var followerProfilePicture string
+
+	// I tried to condesnse this into the insertion but im unsure about syntax,
+	// if its possible to do it in one query some help on that would be appreciated!
```
Comment (ddusichka) [inline backend/internal/storage/postgres/schema/user.go:98]: you can add a `RETURNING` clause to the insert statement above! 

you can look at the UpdateReview method as an example
in `backend/internal/storage/postgres/schema/review.go`

File: backend/internal/supabase/migrations/20241127192734_notifications.sql [hunk 5]
```diff
@@ -0,0 +1,12 @@
+create type notif_type as enum ('follow', 'comment', 'upvote');
+
+CREATE TABLE notifications (
+    id SERIAL PRIMARY KEY,
+    type notif_type NOT NULL,
+    created_at timestamp with time zone default now(),
+    reciever_id uuid NOT NULL references "user"(id),
```
Comment (ddusichka) [inline backend/internal/supabase/migrations/20241127192734_notifications.sql:7]:
```suggestion
    receiver_id uuid NOT NULL references "user"(id),    
```
make sure to update spelling in SQL queries too

File: frontend/app/(tabs)/index.tsx [hunk 6]
```diff
@@ -42,7 +42,7 @@ export default function HomeScreen() {
 
   const handleNotifPress = () => {
     console.log("Notif icon pressed");
```
Comment (ddusichka) [inline frontend/app/(tabs)/index.tsx:44]:
```suggestion
```

**Review by ddusichka — COMMENTED** [review]
YAYYY lots of cool stuff thank you!!!
not sure if this was ready but left some quick comments

also if you find any other things you wanna add/improve would love if you could put that in a new PR just cause this one's getting big

File: backend/internal/models/user.go (hunk 1 above)
Comment (abhikaboy) [reply backend/internal/models/user.go:19]: i'm thinking to leave it flexible for now bc im not sure how we wanna branch this out- but usually from either a user pfp (if its like a follow) or maybe album cover (if the notification is regaurding a review). Unsure though - thoughts?

File: backend/internal/models/user.go (hunk 2 above)
Comment (abhikaboy) [reply backend/internal/models/user.go:22]: oops

Comment (abhikaboy) [reply backend/internal/models/user.go:22]: i before e except after c

File: backend/internal/models/user.go (hunk 1 above)
Comment (ddusichka) [reply backend/internal/models/user.go:19]: ok that makes sense! my only fear here would be a user changing their pfp and it not being updated or something - so maybe the URL should be populated on the fly from the getNotifications endpoint (like if it's type follow, we get the follower's current PFP, if it's something related to a track/album we get the cover, etc.)

not sure if it's worth the extra lookup time though but just my thoughts

File: backend/internal/storage/postgres/storage.go [hunk 7]
```diff
@@ -12,7 +12,7 @@ import (
 )
 
 func ConnectDatabase(config config.DB) *pgxpool.Pool {
-	dbConfig, err := pgxpool.ParseConfig(config.Connection())
+	dbConfig, err := pgxpool.ParseConfig("postgresql://postgres:postgres@127.0.0.1:54322/postgres")
```
Comment (ddusichka) [inline backend/internal/storage/postgres/storage.go:15]: remember to switch this back !

Comment (abhikaboy) [reply backend/internal/storage/postgres/storage.go:15]: You spitting

**Review by ddusichka — APPROVED** [review]
looks good to me!! my suggestions are kind of annoying and none are blockers so feel free to merge as is and we can revisit some of the small things in a followup PR

**Pull request merged**